backend/webserver/tests.py

```diff
@@ -620,3 +623,621 @@ def test_delete_404(self):
+            visibility= "PUBLIC"
+        )
+        url = f'/api/authors/{author_1.id}/posts/{post_1.id}/'
+        self.client.force_authenticate(user=author_1)
+        payload = {
+            "title": "Mark McGoey",
+            "description": "new description",
+            "unlisted":True,
+            "content":"Some new content"
+        }
+        response = self.client.post(url,data=payload)
+        self.assertEqual(response.data["title"], "Mark McGoey")
+        self.assertEqual(response.data["description"], "new description")
+        self.assertEqual(response.data["unlisted"], True)
+        self.assertEqual(response.data["content"], "Some new content")
+
+
+    def test_non_editable_fields(self):
+        author_1 = Author.objects.create() 
+        current_date_string = datetime.datetime.utcnow().replace(tzinfo=utc)
+        post_1 = Post.objects.create(
+            author =author_1,
+            created_at=current_date_string,
+            edited_at=current_date_string,
+            title="Test Post",
+            description="Testing post",
+            source="source",
+            origin="origin",
+            unlisted=False,
+            content_type= "text/plain",
+            content="Some content",
+            visibility= "PUBLIC"
+        )
+        url = f'/api/authors/{author_1.id}/posts/{post_1.id}/'
+        self.client.force_authenticate(user=mock.Mock())
+        payload = {
+            "source":"new source",
+            "origin": "new origin",  
+            "visibility":"FRIENDS"  
+        }
+        self.client.post(url,data=payload)
+        response = self.client.get(url,format=json)
+        
+        self.assertEqual(response.data["origin"], "origin")
+        self.assertEqual(response.data["visibility"], "PUBLIC")
+        self.assertEqual(response.data["source"], "source")
+        
+    
+
+    def test_delete(self):
+        author_1 = Author.objects.create(username="author_1", display_name="author_1")
+        author_1.set_password("pass123")
+        author_1.save()
+        request_payload = {"username": "author_1", "password": "pass123"}
+        self.client.post("/login/", data=request_payload, format="json")
+        current_date_string = datetime.datetime.utcnow().replace(tzinfo=utc)
+        post_1 = Post.objects.create(
+            author =author_1,
+            created_at=current_date_string,
+            title="Test Post",
+            description="Testing post",
+            source="source",
+            origin="origin",
+            unlisted=False,
+            visibility="PUBLIC"
+        )
+        url = f'/api/authors/{author_1.id}/posts/{post_1.id}/'
+        self.client.force_authenticate(user=author_1)
+        self.assertEqual(Post.objects.filter(author=author_1.id).count(),1)
+        response = self.client.delete(url)
+        self.assertEqual(status.HTTP_200_OK, response.status_code)
+        self.assertEqual(Post.objects.filter(author=author_1.id).count(),0)
```

Comment:
```suggestion
        self.assertEqual(0, Post.objects.count())
```
You don't need to filter for post created by author_1.id. You are only creating one post in this test case. So just getting the posts count is enough.